tests/variable_test.py

```diff
@@ -907,6 +907,37 @@ def test_sort():
     assert_export(sc.allsorted, x=var, dim='x', order='ascending')
 
 
+def test_sort_datetime():
+    var = sc.Variable(dims=["x"],
+                      values=[
+                          np.datetime64('2001-01-01T00:00:00', 's'),
+                          np.datetime64('2003-01-01T00:00:00', 's'),
```

Comment:
Do we want/need this test? I assume we have tests of `time_point::operator<`, so what value is added by this?

The only thing we want to somehow ensure is that `sort` works with the dtypes we need. This is actually a generic shortcoming (we do not have any such tests, with a few exceptions). On the other hand, once a type is added to the transform template, such tests would mostly serve as regression tests?

Reply:
I don't see any major disadvantage to testing this functionality... it runs fast and is pretty simple so I don't see it causing maintainability issues. But if you want me to remove it I can do.

Reply:
In my experience it does risk adding maintainability issues. What do you think @jl-wynen?

Reply:
Can you give an example?
I would have thought that any test that states intended behaviour is good. We want sort to work with datetime, so we should have a test which states that it does work.

Reply:
I think I am just wary because in the past we frequently had plenty of tests for the "obvious" things (that are basically guaranteed to work), but the reader got drowned in long code and could not tell whether some actual critical test were missing.

Reply:
> A different point of view: Why test time_point, but not an exhaustive list of all other supported types?

I would have tested all dtypes.
We could change the `assert_export` checks to iterate through all (relevant) dtypes and check if the function accepts those. (At least for unary functions, testing all possible combinations of dtypes in n-ary functions might be overkill.)

Reply:
👍 on testing supported types, provided that we do not attempt to verify values in all cases (for simple transform-based operations where the kernel is tested independently).

Reply:
I would think in principle we should have tests for all the dtypes that `sort` supports. I only added the datetime test in this instance as that's all my PR changed, but I'd be happy to parameterize the test to check for all supported dtypes?

Reply:
> I would think in principle we should have tests for all the dtypes that sort supports.

Can you elaborate why? I feel there are other areas that probably lack significantly in testing and would benefit much more from investing time there. Here we are not going to discover any bugs (related to dtype), and if we do, it should in my opinion be tests of `transform`).

Reply:
I simply feel that, at least in principle, we should have unit tests covering all user-facing functionality that we expose. I am wary of claiming that code is "too obvious to ever break", because in my experience, that's precisely the code that breaks most often...

But I don't really think any of this is important enough to warrant an extended discussion. What would you like me to do with this PR?

Reply:
> I simply feel that, at least in principle, we should have unit tests covering all user-facing functionality that we expose. I am wary of claiming that code is "too obvious to ever break", because in my experience, that's precisely the code that breaks most often...

I fully agree with the "at least in principle". But we have an order of magnitude too few resources for that, so I would prefer to put resources where things are most likely to break.

The way scipp organizes the code using `transform` (with kernels defined and tested in `core`) was *designed* to reduce testing overhead (based on a workshop with the dev team at the time). 3 years ago we had a lot of tests like the above: Many operations were tested based on `Variable`, retesting parts (but never all) of what `transform` does. This had then led to major issues as changes were made (in particular most tests did not test units, since they were not ubiquitous at the time, I believe), and massive refactoring effort going into refactoring tests when the API for constructing variables was changed.